consistency/tool/comparison/comparison_test.py

```diff
@@ -3,34 +3,130 @@
+    word_group_pb2 as wg,
+)
+from google.cloud.apigeeregistry.applications.v1alpha1.consistency import (
+    consistency_report_pb2 as cr,
+)
+from google.protobuf.json_format import ParseDict
+from strsimpy import SorensenDice
 class TestComparison(unittest.TestCase):
     @parameterized.expand(
-        ["simple", "none-ids", "none-words", "unique-terms", "both-null"]
+        [
+            "simple",
+            "none-wordgroups",
+            "none-words",
+            "both-null",
+        ]  # , "unique-terms", "both-null"]
     )
-    @patch.object(Comparison, "get_word_group_ids")
-    def test_find_word_groups(self, name, mock_get_word_group_ids):
+    def test_find_word_groups(
+        self,
+        name,
+    ):
 
         # PATCH
         # Construct mock_response
         with open("comparison_test.json", "r") as myfile:
             data = myfile.read()
         obj = json.loads(data)
         test_suite = obj[name]
-        ids = test_suite["ids"]
-        new_words = test_suite["words"]
-        expected = test_suite["closest_word_groups"]
-
-        mock_get_word_group_ids.return_value = ids
+        input_wordgroups = []
+        if test_suite["wordgroups"] == None or test_suite["words"] == None:
+            return None
+        for i in test_suite["wordgroups"]:
+            wrd_grp = wg.WordGroup()
+            if i != None:
+                input_wordgroups.append(ParseDict(i, wrd_grp))
+            else:
+                input_wordgroups.append(None)
 
         # CALL
-        cmprsn = Comparison(stub="stub", new_words=new_words)
-        actual = cmprsn.find_word_groups()
+        cmprsn = Comparison(
+            stub="stub", new_words=test_suite["words"], word_groups=input_wordgroups
+        )
+        actual = cmprsn.find_closest_word_groups()
+        expected = {}
+        for word, comparison_info in test_suite["expected"].items():
+            wrd_grp = wg.WordGroup()
+            ParseDict(comparison_info[0], wrd_grp)
+            expected[word] = [wrd_grp, comparison_info[1]]
 
         # ASSERT
-        self.assertEqual(actual, expected)
+        self.assertDictEqual(actual, expected)
+
+    @parameterized.expand(["unique-terms"])  # , "unique-terms", "both-null"]
+    def test_find_word_groups_unique(
+        self,
+        name,
+    ):
+
+        # PATCH
+        # Construct mock_response
+        with open("comparison_test.json", "r") as myfile:
+            data = myfile.read()
+        obj = json.loads(data)
+        test_suite = obj[name]
+        input_wordgroups = []
+        if test_suite["wordgroups"] == None or test_suite["words"] == None:
+            return None
+        for i in test_suite["wordgroups"]:
+            wrd_grp = wg.WordGroup()
+            if i != None:
+                input_wordgroups.append(ParseDict(i, wrd_grp))
+            else:
+                input_wordgroups.append(None)
+
+        # CALL
+        cmprsn = Comparison(
```

Comment:
In the unique terms case, since the comparison info will add noise group by default, can we make sure we initialize `Comparison` with a dummy set of noise WordGroup? And the in your test's expected response, you can add that instead of `null`.